File: kora-app-annotation-processor/src/test/java/io/koraframework/kora/app/annotation/processor/app/AppWithFactories10.java
package io.koraframework.kora.app.annotation.processor.app;

import io.koraframework.common.KoraApp;
import io.koraframework.common.annotation.Root;

import java.io.Closeable;

@KoraApp
public interface AppWithFactories10 {
    @Root
    default Object mock1(Closeable object) {
        return new Object();
    }
}

File: kora-app-symbol-processor/src/test/kotlin/io/koraframework/kora/app/ksp/app/AppWithFactories10.kt
package io.koraframework.kora.app.ksp.app

import io.koraframework.common.KoraApp
import io.koraframework.common.annotation.Root
import java.io.Closeable

@KoraApp
interface AppWithFactories10 {
    @Root
    fun mock1(registry: Closeable) = Any()
}
